src/network/message_filter.rs

```diff
@@ -0,0 +1,76 @@
+//!
+//! BIP157  Client Side Block Filtering network messages
+//!
+use util::hash::Sha256dHash;
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+/// getcfilters message
+pub struct GetCFilters {
+    /// Filter type for which headers are requested
+    pub filter_type: u8,
+    /// The height of the first block in the requested range
+    pub start_height: u32,
+    /// The hash of the last block in the requested range
+    pub stop_hash: Sha256dHash,
+}
+impl_consensus_encoding!(GetCFilters, filter_type, start_height, stop_hash);
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+/// cfilter message
+pub struct CFilter {
```

Comment:
Btw, on cfilter message, reading the BIP, "The FilterType SHOULD match the field in the getcfilters request, and BlockHash must correspond to a block that is an ancestor of StopHash with height greater than or equal to StartHeight.". A nit, but I think there the BIP is ambiguous, BlockHash could be also a block hash that is StopHash itself so not only its ancestor ?

Reply:
all comments are copy-paste from BIP157 tables, not my language.

Reply:
This one isn't an issue in PR but one in the BIP itself, I was curious to have your advice on it ?